### PR TITLE
Add "Record Status" filter options to roster redesign

### PR DESCRIPTION
**Pull request by lukenewman**
## Background
Add in "Complete" and "Incomplete" options to the "Record Status" filter. "Complete" will show only records with no validation errors, while "Incomplete" will show only records _with_ validation errors.

## GitHub Issue
https://github.com/ctoec/data-collection/issues/1470

## Associated PRs
If applicable, replace this text with links to additional pull requests associated with this work.

## Validation Plan
- [ ] Have a roster with children in it
- [ ] Filter by "Complete" -- you should see only children with no validation errors
- [ ] Filter by "Incomplete" -- you should see only children _with_ validation errors. If you have none, you can edit a record's enrollment to get some validation errors.

## Tech Specs
- [ ] If any TypeORM entities were created/updated/removed, the necessary migration(s) were [correctly auto-generated](https://github.com/typeorm/typeorm/blob/master/docs/using-cli.md#generate-a-migration-from-existing-table-schema) and **not manually written**.

## Automated Testing
- [ ] All unit tests are passing.
- [ ] Unit tests have been added to cover this changeset.

## Additional Context
If there's any additional information that those reviewing or validating this work should be aware of, please specify it here.

### CODE REVIEW

**Comment by lukenewman** [comment]
One product question I have regarding this and other filters, should we be considering these to be "subtractive" filters or "additive" filters? For example, if you go select all the funding spaces, no children will show up because no record is a member of every funding space. I consider this the "subtractive" filter model. In an "additive" model, this situation would show all children because any record with a funding space will be included in the selected options.

Regarding this particular filter, it almost seems like a toggle may be an option, i.e. "Show Incomplete Records Only". If you go select both "Complete" and "Incomplete" nothing will show up, which I feel like is an unnecessary user path. Happy to talk about this "irl" cause it's not ideal to describe in text.

**Review by jessethach — APPROVED** [review]

**Review by gordonfarrell — APPROVED** [review]
Yay filters! Looks good

**Comment by jessethach** [comment]
I can validate this.

File: src/controllers/children/get.ts [hunk 1]
```diff
@@ -224,6 +224,10 @@ export const getChildren = async (
   const [children, totalItems] = await qb.getManyAndCount();
   let validatedChildren = await Promise.all(children.map(validateChild));
   let { ...filterSet } = query ?? {};
+  /// we can't filter by validation status in `getChildrenQuery`
+  /// so we keep track of a separate total count to return
+  /// the appropriate total pages and count
+  var totalChildren = totalItems;
```
Comment (jessethach) [inline src/controllers/children/get.ts:230]: This should be a `let` probably.

Comment (lukenewman) [reply src/controllers/children/get.ts]: fixed (my swift is showing)

**Comment by lukenewman** [comment]
@jessethach this is ready for another validation when you have a moment

**Comment by jessethach** [comment]
This is validated with most recent changes! @nickbristow Good to merge.

**Pull request merged**